app/controllers/guaranteeDetails/ReferenceNumberController.scala

```diff
@@ -62,7 +62,7 @@ class ReferenceNumberController @Inject() (
           formWithErrors => Future.successful(BadRequest(view(formWithErrors, lrn, mode, index))),
           value => {
             implicit val navigator: GuaranteeNavigator = navigatorProvider(index)
-            ReferenceNumberPage(index).writeToUserAnswers(value).writeToSession().navigateWith(mode)
+            ReferenceNumberPage(index).writeToUserAnswers(value.replaceAll("\\s", "")).writeToSession().navigateWith(mode)
```

Comment:
Does calling textWithSpacesRemoved in the form not ensure that this is already done?

Also it would be good to see a unit test in the ReferenceNumberFormProviderSpec. There's an example in EoriNumberFormProviderSpec.

Reply:
Done